backend/src/main/java/com/google/rolecall/services/SectionServices.java

```diff
@@ -340,6 +305,29 @@ public void deleteSection(int id) throws EntityNotFoundException, InvalidParamet
     sectionRepo.deleteById(id);
   }
 
+  // Utility functions
+
+  private void UpdateRevelationChildren(Section section, Integer[] siblingIndexArray,
+      boolean isParentRevelation) throws InvalidParameterException {
+    if(isParentRevelation) {
+      // Update sibling Ballets with ids of Revelation's internal Ballet/Position structures
+      int loopCounter = 0;
+      for(Position pos: section.getPositions()) {
```

Comment:
If we have for-each loops with external counters, why not just use a traditional for-loop?

Reply:
Fixed. However, if the items were stored in a linked list, the old solution would be better.